spec/features/company_pages_spec.rb

```diff
@@ -0,0 +1,141 @@
+    end
+
+    scenario "it should have an add new company button" do
+      visit companies_path
+      expect(page).to have_content "+ New Company"
+    end
+  end
+end
+
+feature "creating a new company" do
+  let(:cohort) { FactoryGirl.create(:cohort) }
+  let(:admin) { FactoryGirl.create(:admin) }
+  before { login_as(admin, scope: :admin) }
+
+
+  context "with valid input" do
+    scenario "it adds a new company" do
+      visit companies_path
+      click_link "+ New Company"
+      fill_in "Company Name", with: "New Company"
+      fill_in "Phone Number", with: "555-555-5555"
+      fill_in "Email", with: "test@test.com"
+      click_on "Create Company"
+      expect(page).to have_content "New Company"
+    end
+  end
+
+  context "with invalid input" do
+    scenario "it adds a new company" do
+      visit companies_path
+      click_link "+ New Company"
+      fill_in "Company Name", with: ""
+      fill_in "Phone Number", with: "555-555-5555"
+      fill_in "Email", with: "test@test.com"
+      click_on "Create Company"
+      expect(page).to have_content "Company Name"
+    end
+  end
+
+  context "not an admin" do
+    it_behaves_like "require admin login" do
+      let(:action) do
+        visit new_company_path
+      end
+    end
+  end
+end
+
+feature "editing a company" do
+  let(:cohort) { FactoryGirl.create(:cohort) }
+  let(:admin) { FactoryGirl.create(:admin) }
+  let!(:company) { FactoryGirl.create(:company) }
+  before { login_as(admin, scope: :admin) }
+
+  context "with valid input" do
+    scenario "edits a company successfully" do
+      visit companies_path
+      click_link "Edit"
+      fill_in "Company Name", with: "New Company Name"
+      click_on "Update Company"
+      expect(page).to have_content "New Company Name"
+    end
+  end
+
+  context "with invalid input" do
+    scenario "edits a company successfully" do
+      visit companies_path
+      click_link "Edit"
+      fill_in "Company Name", with: ""
+      click_on "Update Company"
+      expect(page).to have_content "Edit Company"
+    end
+  end
+
+  context "not an admin" do
+    let!(:company) { FactoryGirl.create(:company) }
+
+    it_behaves_like "require admin login" do
+      let(:action) { visit edit_company_path(company) }
+    end
+  end
+end
+
+feature "deleting a company" do
+  let(:cohort) { FactoryGirl.create(:cohort) }
+  let(:admin) { FactoryGirl.create(:admin) }
+  let!(:company) { FactoryGirl.create(:company) }
+  before { login_as(admin, scope: :admin) }
+
+  context "as an admin" do
+    scenario "it removes company from database" do
+      visit companies_path
+      click_link "Delete"
+      expect(page).to have_content "Companies"
+      expect(page).to_not have_content company.name
+    end
+  end
+
+  context "not an admin" do
+    let!(:company) { FactoryGirl.create(:company) }
+
+    it_behaves_like "require admin login" do
+      let(:action) { page.driver.submit :delete, "/companies/#{company.id}", {} }
```

Comment:
we aren't really doing these tests of making HTTP requests that the user can't navigate to. increasingly, i don't like our authorization testing approach. we can talk about it more if it's something you're interested in.